src/code_80005FD0.c

```diff
@@ -1316,65 +1343,64 @@ s32 func_80008E58(s32 payerId, s32 pathIndex) {
     posZ = player->pos[2];
     stackPadding = pathIndex;
     trackSectionId = get_track_section_id(player->collision.meshIndexZX);
-    D_80163318[payerId] = trackSectionId;
+    gPlayersTrackSectionId[payerId] = trackSectionId;
     sSomeNearestWaypoint = find_closest_waypoint_track_section(posX, posY, posZ, trackSectionId, &pathIndex);
     gNearestWaypointByPlayerId[payerId] = sSomeNearestWaypoint;
     if (pathIndex) {};
     gPathIndexByPlayerId[payerId] = pathIndex;
-    func_80008DC0(pathIndex);
+    set_current_path(pathIndex);
     if (stackPadding) {};
     return pathIndex;
 }
 
-void func_80008F38(s32 playerId) {
-    f32 temp_f0;
+void update_player_completion(s32 playerId) {
```

Comment:
Might want to add a comment if it uses percentages.

// Completion is tracked using percentages
or something like that.